core/js/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+pub mod utils;
+mod models;
+use models::*;
+
+mod prelude {
+  pub use wasm_bindgen_futures::JsFuture;
+  pub use async_trait::async_trait;
+  pub use wasm_bindgen::prelude::*;
+  pub use web_sys::{ window, CredentialCreationOptions, CredentialRequestOptions, PublicKeyCredentialCreationOptions, PublicKeyCredentialRequestOptions, PublicKeyCredentialRpEntity, PublicKeyCredentialUserEntity };
+  pub use serde::{ Serialize, Serializer, Deserialize, Deserializer, de::{ SeqAccess, Visitor, Error as DeError } };
+  pub use js_sys::Uint8Array;
+  pub use virto_sdk::{ authenticator::AuthError, signer::SignerError, AuthResult, Authenticator, Signer, SignerResult };
+  pub use crate::utils;
+}
+
+use prelude::*;
+
+pub struct Profile<'a> {
+  username: &'a str,
+  display_name: &'a str,
+}
+pub struct Credentials<'a> {
+  username: &'a str,
+}
+
+#[wasm_bindgen]
+pub struct WebAuthN  {
+  rp_id: String,
+} 
+
+#[async_trait(?Send)]
+impl Authenticator for WebAuthN {
+  type Credentials<'c> = Credentials<'c>;
+  type Profile<'p> = Profile<'p>;
+  type RegResponse = WebAuthnResult<RegResponse>;
+
+  async fn register<'m>(&self, profile: &'m Profile<'_>) -> AuthResult<Self::RegResponse> {
+    let window = window().expect("expect window to exist");
```

Comment:
Notes on [defensive programming](https://polkadot-blockchain-academy.github.io/pba-book/substrate/tips-tricks/page.html?highlight=qed#defensive-programming): you ONLY `expect` a value when you are **certain** such value is ensured.

Otherwise, return a regular error instead.

Remember: `expect` panics if `Ok` is not fulfilled. An user might potentially want to use this libary (for other reasons) on a non-navigator environment (a js runtime, for example).

```suggestion
    let window = window().ok_or(AuthError::NoWindow)?;
```

Alternatively, try to somehow, inject the `credentials_manager` (as a parameter, maybe).